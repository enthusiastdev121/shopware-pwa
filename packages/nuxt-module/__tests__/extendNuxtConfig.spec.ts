import extendNuxtConfig from "../src/extendNuxtConfig";

describe("nuxt-module - extendNuxtConfig", () => {
  it("should return default config", () => {
    const result = extendNuxtConfig({});
    expect(result).toMatchSnapshot();
  });

  it("should overwrite server port", () => {
    const result = extendNuxtConfig({
      server: {
        port: 2000,
      },
    });
    expect(result.server?.port).toEqual(2000);
  });

  it("should add new build module", () => {
    const result = extendNuxtConfig({
      buildModules: ["some-new-module"],
    });
    expect(result.buildModules).toEqual([
      "@nuxt/typescript-build",
      "@shopware-pwa/nuxt-module",
      "some-new-module",
    ]);
  });

  describe("env properties", () => {
    const ORIGINAL_ENV = process.env;

    beforeEach(() => {
      jest.resetModules(); // clears the cache
      process.env = { ...ORIGINAL_ENV };
    });

    afterAll(() => {
      process.env = ORIGINAL_ENV;
    });

    it("should pass env properties to config", () => {
      process.env.NODE_ENV = "production";
      process.env.EXPERIMENTAL_IMAGE_PROCESSING_SERVER = "https://example.com";

      const extendNuxtConfig = require("../src/extendNuxtConfig").default;
      const result = extendNuxtConfig({});

      expect(result.env?.CHOKIDAR_USEPOLLING).toEqual("0");
      expect(result.env?.EXPERIMENTAL_IMAGE_PROCESSING_SERVER).toEqual(
        "https://example.com"
      );
    });
  });
});
