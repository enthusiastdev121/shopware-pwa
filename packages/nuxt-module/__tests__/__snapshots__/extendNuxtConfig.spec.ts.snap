// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nuxt-module - extendNuxtConfig should return default config 1`] = `
Object {
  "axios": Object {},
  "build": Object {
    "babel": Object {
      "plugins": Array [
        Array [
          "@babel/plugin-proposal-private-methods",
          Object {
            "loose": true,
          },
        ],
      ],
    },
  },
  "buildModules": Array [
    "@nuxt/typescript-build",
    "@shopware-pwa/nuxt-module",
  ],
  "components": false,
  "css": Array [
    "~assets/scss/main.scss",
  ],
  "env": Object {
    "CHOKIDAR_USEPOLLING": "1",
    "EXPERIMENTAL_IMAGE_PROCESSING_SERVER": "",
  },
  "head": Object {
    "link": Array [
      Object {
        "href": "/favicon.ico",
        "rel": "icon",
        "type": "image/x-icon",
      },
      Object {
        "as": "style",
        "href": "https://fonts.googleapis.com/css?family=Raleway:300,400,400i,500,600,700%7CRoboto:300,300i,400,400i,500,700&display=swap",
        "rel": "preload",
      },
      Object {
        "href": "https://fonts.googleapis.com/css?family=Raleway:300,400,400i,500,600,700%7CRoboto:300,300i,400,400i,500,700&display=swap",
        "media": "print",
        "once": true,
        "onload": "this.media='all'",
        "rel": "stylesheet",
      },
    ],
    "meta": Array [
      Object {
        "content": "shopware-pwa",
        "hid": "project-type",
        "name": "project-type",
      },
      Object {
        "charset": "utf-8",
      },
      Object {
        "content": "width=device-width, initial-scale=1",
        "name": "viewport",
      },
      Object {
        "content": "",
        "hid": "description",
        "name": "description",
      },
    ],
    "title": "Shopware PWA",
  },
  "modules": Array [
    "@nuxtjs/axios",
    "@nuxtjs/pwa",
  ],
  "plugins": Array [],
  "pwa": Object {
    "manifest": Object {
      "lang": "en",
    },
    "workbox": Object {
      "offlineStrategy": "StaleWhileRevalidate",
    },
  },
  "server": Object {
    "host": "0.0.0.0",
    "port": 3000,
  },
  "telemetry": false,
}
`;
